src/realm.h

```diff
@@ -1235,6 +1243,20 @@ RLM_API realm_object_t* realm_object_create_with_primary_key(realm_t*, realm_cla
  */
 RLM_API bool realm_object_delete(realm_object_t*);
 
+/**
+ * Freeze the Realm object in the provided Realm.
```

Comment:
Might be worth clarifying that this is exactly equivalent to producing a thread-safe reference and resolving it in the frozen realm (and conversely for the "thaw" methods). If this is not actually the case, explain the difference 😂